zebra/zebra_router.h

```diff
@@ -71,6 +71,35 @@ struct zebra_mlag_info {
 
 	/* The system mac being used */
 	struct ethaddr mac;
+	/*
+	 * Zebra will open the communication channel with MLAGD only if any
+	 * clients are interested and it is controlled dynamically based on
+	 * client registers & un-registers.
+	 */
+	uint32_t clients_interested_cnt;
+
+	/* coomunication channel with MLAGD is established */
+	bool connected;
+
+	/* connection retry timer is running */
+	bool timer_running;
+
+	/* Holds the client data(unencoded) that need to be pushed to MCLAGD*/
+	struct stream_fifo *mlag_fifo;
+
+	/*
+	 * A new Kernel thread will be created to post the data to MCLAGD.
+	 * where as, read will be performed from the zebra main thread, because
+	 * read involves accessing client registartion data structures.
+	 */
+	struct frr_pthread *zebra_pth_mlag;
+
+	/* MLAG Thread context 'master' */
+	struct thread_master *th_master;
+
+	/* Threads for read/write. */
+	struct thread *t_read;
```

Comment:
Yes, because at any point of  time only one event can be possible (out of these 3)

1) Initially this event is used to connect to MLAGD, one time operation with FRR Client Registration
2) In case connection failed in first attempt, Timer retry will kick in, this no time no other 2 events can be scheduled.
3) only after successful connection establishment, Read can be performed.

all 3 are Mutually exclusive,

Reply:
so ... in the interest of self-documenting code, it would be helpful a) not to have a comment that says "for read/write", and b) to have the comment that says clearly "this is used in these three ways, and here's why that is ok (for now)." I mean, you understand the issue, right? there are lots of places that use the 'struct thread *' to ensure that a task is singleton - but you're doing a different thing here.

Reply:
added Comments to explain the usage.